### PR TITLE
Theme provider

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>

- originally created theme with CssVarsProvider but backtracked to switch all material MUI components into Joy MUI components
- Implemented use of react components in error state component
- Switched TextField components into FormControl, FormLabel, and Input elements

<what will be done in later PRs and not included here>

- utilize CssVarsProvider to have consistent styling

## Milestones
<the milestones or stories/features that this works towards>

- refactoring code into react components

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>
https://mui.com/joy-ui/customization/approaches/
https://mui.com/joy-ui/react-input/
https://mui.com/joy-ui/react-alert/
https://mui.com/material-ui/material-icons/

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
Images located in google document
https://docs.google.com/document/d/1u4QuX_0hLM_7-EKXGbdEtpxEErOixgLAIE5qFVQna7s/edit?usp=sharing

### CODE REVIEW

**Review by nithyad — APPROVED** [review]
LGTM

File: frontend/src/components/ErrorState.tsx [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+import ReportIcon from "@mui/icons-material/Report";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
 import type { GPErrorMessageTypes } from "../utils/types";
 
 const ErrorState: React.FC<GPErrorMessageTypes> = ({ error, message }) => {
   const displayedMessage = errorCodes[message] ?? message;
   return (
     <div>
-      <Alert severity={error ? "error" : "success"}>{displayedMessage}</Alert>
+      <Alert
```
Comment (nithyad) [inline frontend/src/components/ErrorState.tsx:12]: nice!

**Review by nithyad — APPROVED** [review]

File: frontend/src/utils/UIStyle.ts [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+const theme = extendTheme({
+  colorSchemes: {
+    light: {
+      palette: {
+        // affects all Joy components that has `color="primary"` prop.
+        primary: {
+          50: "#fffbeb",
+          100: "#fef3c7",
+          200: "#fde68a",
+          // 300, 400, ..., 800,
```
Comment (nithyad) [inline frontend/src/utils/UIStyle.ts:12]: not sure if this comment is needed?
